StaticHTMLRenderer cannot handle ValidationError

## Checklist
- [ ] Raised initially as discussion #...
- [x] This cannot be dealt with as a third party library. (We prefer new functionality to be [in the form of third party libraries](https://www.django-rest-framework.org/community/third-party-packages/#about-third-party-packages) where possible.)
- [x] I have reduced the issue to the simplest possible case.

## Description
If any `ValidationError` is raised in a view which is going to use `StaticHTMLRenderer` then it will cause an error resulting in an Internal-Server-Error because the base class `TemplateHTMLRenderer` expectes to handle with a `dict` while in this case it receives a `list` instead.

https://github.com/encode/django-rest-framework/blob/0f39e0124d358b0098261f070175fa8e0359b739/rest_framework/renderers.py#L171-L175

## Code sample

```python
from django.test import TestCase
from rest_framework import exceptions
from rest_framework.renderers import StaticHTMLRenderer
from rest_framework.test import URLPatternsTestCase


class StaticHTMLRendererErrorTests(URLPatternsTestCase):
    class StaticErrorView(APIView):
        renderer_classes = (StaticHTMLRenderer,)

        def get(self, request, **kwargs):
            raise exceptions.ValidationError('error')

    urlpatterns = [path('error', StaticErrorView.as_view())]

    def test_statis_renderer_with_api_exception(self):
        response = self.client.get('/error')  # this line raise a TypeError
        assert response.status_code == 400
```

### Stacktrace
<details>
<summary><code>TypeError: list indices must be integers or slices, not str</code></summary>

```
failed: tests/test_renderers.py:633: in test_statis_renderer_with_api_exception
    response = self.client.get('/error')
.venv/lib/python/site-packages/django/test/client.py:742: in get
    response = super().get(path, data=data, secure=secure, **extra)
.venv/lib/python/site-packages/django/test/client.py:398: in get
    **extra,
.venv/lib/python/site-packages/django/test/client.py:473: in generic
    return self.request(**r)
.venv/lib/python/site-packages/django/test/client.py:714: in request
    response = self.handler(environ)
.venv/lib/python/site-packages/django/test/client.py:145: in __call__
    response = self.get_response(request)
.venv/lib/python/site-packages/django/core/handlers/base.py:130: in get_response
    response = self._middleware_chain(request)
.venv/lib/python/site-packages/django/core/handlers/exception.py:49: in inner
    response = response_for_exception(request, exc)
.venv/lib/python/site-packages/django/core/handlers/exception.py:114: in response_for_exception
    response = handle_uncaught_exception(request, get_resolver(get_urlconf()), sys.exc_info())
.venv/lib/python/site-packages/django/core/handlers/exception.py:47: in inner
    response = get_response(request)
.venv/lib/python/site-packages/django/utils/deprecation.py:117: in __call__
    response = response or self.get_response(request)
.venv/lib/python/site-packages/django/core/handlers/exception.py:49: in inner
    response = response_for_exception(request, exc)
.venv/lib/python/site-packages/django/core/handlers/exception.py:114: in response_for_exception
    response = handle_uncaught_exception(request, get_resolver(get_urlconf()), sys.exc_info())
.venv/lib/python/site-packages/django/core/handlers/exception.py:47: in inner
    response = get_response(request)
.venv/lib/python/site-packages/django/utils/deprecation.py:117: in __call__
    response = response or self.get_response(request)
.venv/lib/python/site-packages/django/core/handlers/exception.py:49: in inner
    response = response_for_exception(request, exc)
.venv/lib/python/site-packages/django/core/handlers/exception.py:114: in response_for_exception
    response = handle_uncaught_exception(request, get_resolver(get_urlconf()), sys.exc_info())
.venv/lib/python/site-packages/django/core/handlers/exception.py:47: in inner
    response = get_response(request)
.venv/lib/python/site-packages/django/utils/deprecation.py:117: in __call__
    response = response or self.get_response(request)
.venv/lib/python/site-packages/django/core/handlers/exception.py:49: in inner
    response = response_for_exception(request, exc)
.venv/lib/python/site-packages/django/core/handlers/exception.py:114: in response_for_exception
    response = handle_uncaught_exception(request, get_resolver(get_urlconf()), sys.exc_info())
.venv/lib/python/site-packages/django/core/handlers/exception.py:47: in inner
    response = get_response(request)
.venv/lib/python/site-packages/django/utils/deprecation.py:117: in __call__
    response = response or self.get_response(request)
.venv/lib/python/site-packages/django/core/handlers/exception.py:49: in inner
    response = response_for_exception(request, exc)
.venv/lib/python/site-packages/django/core/handlers/exception.py:114: in response_for_exception
    response = handle_uncaught_exception(request, get_resolver(get_urlconf()), sys.exc_info())
.venv/lib/python/site-packages/django/core/handlers/exception.py:47: in inner
    response = get_response(request)
.venv/lib/python/site-packages/django/core/handlers/base.py:204: in _get_response
    response = response.render()
.venv/lib/python/site-packages/django/template/response.py:105: in render
    self.content = self.rendered_content
rest_framework/response.py:74: in rendered_content
    ret = renderer.render(self.data, accepted_media_type, context)
rest_framework/renderers.py:232: in render
    context = self.get_template_context(data, renderer_context)
rest_framework/renderers.py:174: in get_template_context
    data['status_code'] = response.status_code
E   TypeError: list indices must be integers or slices, not str
```
</details>

--- a/rest_framework/renderers.py
+++ b/rest_framework/renderers.py
@@ -171,6 +171,10 @@ class TemplateHTMLRenderer(BaseRenderer):
 
     def get_template_context(self, data, renderer_context):
         response = renderer_context['response']
+        # in case a ValidationError is caught the data parameter may be a list
+        # see rest_framework.views.exception_handler
+        if isinstance(data, list):
+            return {'details': data, 'status_code': response.status_code}
         if response.exception:
             data['status_code'] = response.status_code
         return data

--- a/tests/test_htmlrenderer.py
+++ b/tests/test_htmlrenderer.py
@@ -8,6 +8,7 @@ from django.urls import path
 
 from rest_framework import status
 from rest_framework.decorators import api_view, renderer_classes
+from rest_framework.exceptions import ValidationError
 from rest_framework.renderers import TemplateHTMLRenderer
 from rest_framework.response import Response
 
@@ -34,10 +35,17 @@ def not_found(request):
     raise Http404()
 
 
+@api_view(('GET',))
+@renderer_classes((TemplateHTMLRenderer,))
+def validation_error(request):
+    raise ValidationError('error')
+
+
 urlpatterns = [
     path('', example),
     path('permission_denied', permission_denied),
     path('not_found', not_found),
+    path('validation_error', validation_error),
 ]
 
 
@@ -89,6 +97,12 @@ class TemplateHTMLRendererTests(TestCase):
         response = self.client.get('/permission_denied')
         self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
         self.assertEqual(response.content, b"403 Forbidden")
+        self.assertEqual(response['Content-Type'], 'text/html; charset=utf-8')
+
+    def test_validation_error_html_view(self):
+        response = self.client.get('/validation_error')
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(response.content, b"400 Bad Request")
         self.assertEqual(response['Content-Type'], 'text/html; charset=utf-8')
 
     # 2 tests below are based on order of if statements in corresponding method

